tools/web3-plugin-example/test/unit/contract_method_wrappers.test.ts

```diff
@@ -0,0 +1,125 @@
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { DEFAULT_RETURN_FORMAT } from 'web3-utils';
+
+import { Web3 } from '../web3_export_helper';
+import { ContractMethodWrappersPlugin } from '../../src/contract_method_wrappers';
+import { ERC20TokenAbi } from '../../src/ERC20Token';
+
+declare module '../web3_export_helper' {
+	interface Web3 {
+		contractMethodWrappersPlugin: ContractMethodWrappersPlugin;
+	}
+}
+
+describe('CustomRpcMethodsPlugin Tests', () => {
+	it('should register CustomRpcMethodsPlugin plugin', () => {
+		const web3 = new Web3('http://127.0.0.1:8545');
+		web3.registerPlugin(
+			new ContractMethodWrappersPlugin(
+				ERC20TokenAbi,
+				'0xdAC17F958D2ee523a2206206994597C13D831ec7',
+			),
+		);
+		expect(web3.contractMethodWrappersPlugin).toBeDefined();
+	});
+
+	describe('CustomRpcMethodsPlugin methods tests', () => {
+		const contractAddress = '0xdAC17F958D2ee523a2206206994597C13D831ec7';
+		const sender = '0x8da5e39ec14b57fb9bcd9aa2b4500e909119795d';
+		const recipient = '0x4f641def1e7845caab95ac717c80416082430d0d';
+		const amount = BigInt(42);
+		const expectedSenderBalance =
+			'0x0000000000000000000000000000000000000000000000000000000000000280';
+		const expectedRecipientBalance =
+			'0x0000000000000000000000000000000000000000000000000000000000000120';
+		const requestManagerSendSpy = jest.fn();
+
+		let web3: Web3;
+
+		beforeAll(() => {
+			web3 = new Web3('http://127.0.0.1:8545');
+			web3.registerPlugin(new ContractMethodWrappersPlugin(ERC20TokenAbi, contractAddress));
+			web3.contractMethodWrappersPlugin._contract.requestManager.send = requestManagerSendSpy;
+		});
+
+		it('should call contractMethodWrappersPlugin.getFormattedBalance with expected RPC object', async () => {
+			requestManagerSendSpy.mockResolvedValueOnce(expectedSenderBalance);
+
+			await web3.contractMethodWrappersPlugin.getFormattedBalance(
+				sender,
+				DEFAULT_RETURN_FORMAT,
+			);
+			expect(requestManagerSendSpy).toHaveBeenCalledWith({
+				method: 'eth_call',
+				params: [
+					{
+						data: '0x70a082310000000000000000000000008da5e39ec14b57fb9bcd9aa2b4500e909119795d',
+						to: '0xdAC17F958D2ee523a2206206994597C13D831ec7',
+					},
+					'latest',
+				],
+			});
+		});
+
+		it('should call CustomRpcMethodsPlugin.customRpcMethodWithParameters with expected RPC object', async () => {
+			const expectedGasPrice = '0x1ca14bd70';
+			const expectedTransactionHash =
+				'0xc41b9a4f654c44552e135f770945916f57c069b80326f9a5f843e613491ab6b1';
+
+			requestManagerSendSpy.mockResolvedValueOnce(expectedGasPrice);
+			// Not sure what's being mocked here
```

Comment:
Could we, please, have an issue for investigating the matter and enhancing this piece of code?

Reply:
Addressed by [this](https://github.com/web3/web3.js/pull/5393/commits/4375a5d1382a7e1417f6cd223924baf684fa4f99) commit